digital_land_frontend/filters.py

```diff
@@ -156,6 +156,29 @@ def filter(self, id, type="name"):
 developer_agreement_mapper = MapperFilter(DeveloperAgreementMapper())
 
 
+class MapperRouter:
+    def __init__(self, mappers):
+        if len(mappers) == 0:
+            raise ValueError("no mappers provided")
+        self.mappers = mappers
+
+    def route(self, k, dataset, type="name"):
+        if dataset in self.mappers:
+            mapper = self.mappers[dataset]
+            return mapper.filter(k, type)
+        else:
```

Comment:
I'd normally flip this if/else around into a guard (like __init__) so it would first check

```
if dataset not in self.mappers:
    raise <Whatever>
```

That guards the rest of the function from executing in case dataset isn't in the lookup. You then don't need to indent the main functionality:

```
return self.mappers[dataset].filter(k, type)
```